test/infrastructure/docker/e2e/docker_test.go

```diff
@@ -236,6 +212,29 @@ var _ = Describe("Docker", func() {
 					return upgraded, nil
 				}, "10m", "30s").Should(Equal(int(*controlPlane.Spec.Replicas)))
 			})
+			deleteClusterInput := framework.DeleteClusterInput{
+				Deleter: client,
+				Cluster: cluster,
+			}
+			framework.DeleteCluster(ctx, deleteClusterInput)
+
+			waitForClusterDeletedInput := framework.WaitForClusterDeletedInput{
+				Getter:  client,
+				Cluster: cluster,
+			}
+			framework.WaitForClusterDeleted(ctx, waitForClusterDeletedInput)
+
+			assertAllClusterAPIResourcesAreGoneInput := framework.AssertAllClusterAPIResourcesAreGoneInput{
+				Lister:  client,
+				Cluster: cluster,
+			}
+			framework.AssertAllClusterAPIResourcesAreGone(ctx, assertAllClusterAPIResourcesAreGoneInput)
+
+			ensureDockerDeletedInput := ensureDockerArtifactsDeletedInput{
+				Lister:  client,
+				Cluster: cluster,
+			}
+			ensureDockerArtifactsDeleted(ensureDockerDeletedInput)
```

Comment:
This will potentially break the tests (or orphan resources) if we eventually want to parallelize the tests. If there are resources that should be shared among all tests, it might be good to move those to BeforeSuite/AfterSuite, which could be shared for all tests running on a given parallel node.